s/core_test.go

```diff
@@ -205,3 +205,50 @@ func TestStartNestedSupervisors(t *testing.T) {
 		)
 	})
 }
+
+func TestStartFailedChild(t *testing.T) {
+	parentName := "root"
+	b0n := "branch0"
+	b1n := "branch1"
+
+	cs := []c.ChildSpec{
+		WaitDoneChild("child0"),
+		WaitDoneChild("child1"),
+		WaitDoneChild("child2"),
+		FailStartChild("child3"),
```

Comment:
can you add something like `// note this failed child`. I chased my tail for a couple of minutes trying to understand why there would be an error.